### ISSUE TITLE
Validate the RandomWalk galaxies with star formation knots

### ISSUE THREAD

**Issue by cwwalter** [LSSTDESC/imSim#131]
@EiffL Added complex morphology galaxies with star formation regions utilizing the GalSim RandomWalk class.  We have instance catalogs made for them and they will be used in run 2.0.

We will produce several focal planes but then validate the output and compare with the pure-serisic cases.

**Comment by rmandelb** [comment]
@cwwalter - is the idea to validate the output by comparing images (which might be quick) or processing with DM?  Does this need Francois's input, or who will be doing this?

**Comment by cwwalter** [comment]
I think we need the input from Francois.  I had assumed we would want to use DM to compare calculated quantities but am interested in his (and your) opinion.

**Comment by EiffL** [comment]
Hi both, Yes I was assuming the same, that we would attempt to validate with the DM outputs. I stand at the ready, to have a look at it as soon as the data is available.

**Comment by EiffL** [comment]
After fixing a number of bugs last week in `sims_GCRCatSimInterface`, I've been able to run a few tests to verify the knots the knots model. Ideally we would have wanted to run a portion of run 1.2i with knots, make a small DRP, and compare the results, but we don't have time for that.

Based on a proposal from @rmandelb , I'm instead doing a more limited comparison of fast ImSim images on a few visits, with and without knots. These tests are verifying the following points:
  - Check that `generateInstCat.py` still works with latest GCR catalogs and knots_addon
  - Check that when `protoDC2_addon_knots` is used, all the knots logic is being triggered, i.e. write knots catalog, and update disks `magnorm`
  - Check that between 2 different visits knots are generated in the same place (they are randomly drawn, but with the same seed that comes from the unique galaxy id)
  - Check that between 2 different filters knots are also being generated in the same place

In order to do these tests, I've built shifter images for CatSim and ImSim, where I've tried to be as close as possible to the software environment on run 2.0 (in terms of package versions). The repo for these images, and the command line I've used to produce instance catalogs and fast ImSim simulations (without sensor module) is here: https://github.com/EiffL/DC2_knots
I've manually extracted from the opsim db the visits with best seeing and depth centered in the middle of protoDC2. Then I've generated the instance catalogs for these visits, and finally, to make the comparison easier, I've manually matched the telescope orientation in all visits, as well as increased the exposure time to 120s and decreased the seeing to 0.001 arcsec (to be able to resolve the knots). Finally, I've ran ImSim in fast mode on these catalogs.
Here is a small example of what that looks like:
![image](https://user-images.githubusercontent.com/861591/45281931-402f5680-b48e-11e8-85c2-d84444859dae.png)
The knots are at the same position between different visits and in different bands, that's good.

It also looks like the flux is correctly preserved (residuals around galaxies sum to values close to zero), but I'm still trying to make a nice plot to show it.

Note also that I had tested a while back the impact of knots on size and flux and found that everything remained consistent (no sign of fluxes not being preserved) (see here: https://github.com/LSSTDESC/descqa/issues/62)

Bottom line, I'm still trying to get a few measurements from the difference images to show that the flux is preserved, but it already looks like that's the case, and everything else seems to be working as expected (with the few bug fixes from last week).

@rmandelb Let me know if there is a specific plot you would like to see.

**Comment by cwwalter** [comment]
Good news.  Thanks @EiffL . Keep us up to date on the flux conservation.

**Comment by EiffL** [comment]
Ok, so I've gone a little further, reading-in the instance catalogs, and extracting cutouts around a random subset of galaxies on one particular sensor visit. 
Here is a cutout:
![image](https://user-images.githubusercontent.com/861591/45344638-1727c800-b559-11e8-899a-3598daeec040.png)
Here is a residual image:
![image](https://user-images.githubusercontent.com/861591/45344653-2444b700-b559-11e8-829c-ef1a6ddec75b.png)
And more interestingly, here is an histogram of the sum of the cutout residuals:
![image](https://user-images.githubusercontent.com/861591/45344506-c3b57a00-b558-11e8-8fd2-c3c483ee26cc.png)
It's unbiased, the scatter comes from noise and other objects at the edges of the stamps. 

@cwwalter I'm pretty convinced by now that everything is working as expected. Just let me know if you have any concerns, otherwise I think it's ok to start instance catalog generation.

**Comment by rmjarvis** [comment]
Looks good to me.

**Comment by cwwalter** [comment]
@EiffL OK that's great.  I am going to close this issue and mark it off the checklist as being validated/complete.

Could you continue to interface with the instance catalog team and do a final check that the catalogs being produced in fact do properly contain the knots?

Thanks again for all this work!

**Comment by EiffL** [comment]
sure, I'll try to check on that
